logical/sanitize.go

```diff
@@ -60,3 +65,46 @@ type HTTPWrapInfo struct {
 	CreationTime    time.Time `json:"creation_time"`
 	WrappedAccessor string    `json:"wrapped_accessor,omitempty"`
 }
+
+type HTTPSysInjector struct {
+	Response *HTTPResponse
+}
+
+func (h HTTPSysInjector) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(h.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fast path no data or empty data
+	if h.Response.Data == nil || len(h.Response.Data) == 0 {
+		return j, nil
+	}
+
+	// Marshaling a response will always be a JSON object, meaning it will
```

Comment:
A valid JSON can also start with a `[`. Just bringing your attention onto this, to be able to cross check if `List` responses are not affected by this (which returns a []string).